pyartcd/pyartcd/oc.py

```diff
@@ -35,3 +40,55 @@ async def registry_login(runtime: Runtime):
     except ChildProcessError:
         runtime.logger.error('Failed to login into OC registry')
         raise
+
+
+def common_oc_wrapper(cmd_result_name: str, cli_verb: str, oc_args: List[str], check_status: bool = False, return_value: bool = False):
+    # cmd_result_name: Result obj name in log
+    # cli_verb: first command group
+    # oc_args: args list of command
+    # check_status: whether check status and print output in log
+    # return_value: whether need return value sets
+    logger.info(f"run: oc {cli_verb} {' '.join(oc_args)}")
+    with oc.tracking() as tracker:
+        try:
+            r = Result(cmd_result_name)
+            r.add_action(oc.oc_action(oc.cur_context(), cli_verb, cmd_args=oc_args))
+            if check_status:
+                if r.status() == 0:
+                    logger.info(f"Output: {r.out().strip()}")
+                else:
+                    logger.warn(r.err())
+            r.fail_if(f"oc action {cmd_result_name} failed")
+        except Exception as e:
+            logger.error(tracker.get_result())
+            raise e
+    if return_value:
+        return r.status(), r.out().strip()
+
+
+def get_release_image_info_from_pullspec(pullspec: str):
+    # oc image info --output=json <pullspec>
+    cmd_args = ['info', "--output=json", pullspec]
+    res, out = common_oc_wrapper("single_image_info", "image", cmd_args, False, True)
+    return res, json.loads(out)
+
+
+def extract_release_binary(image_pullspec: str, path_args: List[str]):
+    # oc image extract --confirm --only-files --path=/usr/bin/..:<workdir> <pullspec>
+    cmd_args = ['extract', '--confirm', '--only-files'] + path_args + [image_pullspec]
+    common_oc_wrapper("extract_image", "image", cmd_args, False, False)
+
+
+def get_release_image_pullspec(release_pullspec: str, image: str):
```

Comment:
Please add type declarations to the output (also of the other functions).

Think the first one should be a `bool` rather than an `int`, as it makes reading the resulting code easier.

Reply:
the result.status is an int value https://github.com/openshift/openshift-client-python/blob/main/packages/openshift/result.py#L29

Reply:
updated